apps/site/assets/ts/schedule/components/direction/ScheduleDirectionMenu.tsx

```diff
@@ -1,44 +1,359 @@
+      onClick={handleClick}
+      ref={item => item && focused && item.focus()}
+      onKeyUp={(e: ReactKeyboardEvent) => {
+        handleReactEnterKeyPress(e, () => {
+          handleClick();
+        });
+      }}
+      onKeyDown={(e: ReactKeyboardEvent) => {
+        handleNavigation(e, routePatternIds);
+      }}
+    >
+      <div className="m-schedule-direction__menu-item-headsign">
+        {icon}
+        {routePattern.headsign}
+      </div>
+      <div className="m-schedule-direction__menu-item-description">
+        {duplicated && `from ${routePattern.name.split(" - ")[0]}, `}
+        {routePattern.time_desc ||
+          typicalityDefaultText(routePattern.typicality)}
+      </div>
+    </div>
+  );
+};
+
+const hasMoreRoutePatterns = (routePatterns: EnhancedRoutePattern[]): boolean =>
+  routePatterns.some(
+    (routePattern: EnhancedRoutePattern) => routePattern.typicality > 2
+  );
+
+const determineFocusIndex = (
+  itemFocus: string | null,
+  routePatterns: EnhancedRoutePattern[]
+): number => {
+  if (itemFocus === "first") return 0;
+  return routePatterns.findIndex(
+    (routePattern: EnhancedRoutePattern) => routePattern.typicality > 2
+  );
+};
+
+const isDuplicateHeadsign = (
+  compareRoutePattern: EnhancedRoutePattern,
+  routePatterns: EnhancedRoutePattern[]
+): boolean =>
+  routePatterns.filter(
+    (routePattern: EnhancedRoutePattern) =>
+      routePattern.id !== compareRoutePattern.id &&
+      routePattern.headsign === compareRoutePattern.headsign
+  ).length > 0;
+
+const ExpandedMenu = ({
+  routePatterns,
+  selectedRoutePatternId,
+  showAllRoutePatterns,
+  itemFocus,
+  dispatch
+}: ExpandedMenuProps): ReactElement<HTMLElement> => {
+  const filterRule = (routePattern: EnhancedRoutePattern): boolean =>
+    showAllRoutePatterns ? true : routePattern.typicality < 3;
+  const handleClick = (): void =>
+    dispatch({ event: "showAllRoutePatterns", payload: {} });
+  const focusIndex = determineFocusIndex(itemFocus, routePatterns);
+  const toggleButtonIds =
+    hasMoreRoutePatterns(routePatterns) && showAllRoutePatterns === false
+      ? ["uncommon"]
+      : [];
+  const routePatternIds = routePatterns
+    .filter(filterRule)
+    .map((routePattern: EnhancedRoutePattern) => routePattern.id)
+    .concat(toggleButtonIds);
+  return (
+    <div className="m-schedule-direction__menu" role="menu">
+      {routePatterns
+        .filter(filterRule)
+        .map((routePattern: EnhancedRoutePattern, index: number) => (
+          <RoutePatternItem
+            key={routePattern.id}
+            routePatternIds={routePatternIds}
+            routePattern={routePattern}
+            selected={selectedRoutePatternId === routePattern.id}
+            focused={index === focusIndex}
+            duplicated={isDuplicateHeadsign(routePattern, routePatterns)}
+            dispatch={dispatch}
+          />
+        ))}
+      {hasMoreRoutePatterns(routePatterns) && showAllRoutePatterns === false && (
```

Comment:
I don't know if this item as a button that gets replaced with extra items could be more accessible -- we could wait on feedback on it from IHCD.
I think for now we could add a hidden div / label that says something like "click for additional routes" or something.

Reply:
good call. Added an `aria-label` and the experience was much better w/ screen reader